pkg/nodediscovery/nodediscovery.go

```diff
@@ -176,7 +176,7 @@ func (n *NodeDiscovery) StartDiscovery(nodeName string) {
 	n.LocalNode.EncryptionKey = node.GetIPsecKeyIdentity()
 	n.LocalNode.WireguardPubKey = node.GetWireguardPubKey()
 	n.LocalNode.Labels = node.GetLabels()
-	n.LocalNode.NodeIdentity = identity.GetLocalNodeID().Uint32()
+	n.LocalNode.NodeIdentity = uint32(identity.ReservedIdentityHost)
```

Comment:
It already has its own PR :) https://github.com/cilium/cilium/pull/17836

As for impact, it's a bit hard to say. One consequence is that internally we will handle the `NodeUpdated()` logic for the local node when we receive an update from k8s, even though the code is intending for this case to _not_ trigger node update (because the same information is learned locally from the `NodeDiscovery` component).

Reply:
Ah, it's in my review queue already. Nice!

Ok, thanks for the explanation. It looks fairly benign, but best to fix and backport I guess :-)

Reply:
Oh, actually I think this could potentially cause the identity of the local host to be considered as `remote-node` which could be bad for policy. Particularly because this is the "local" node case which will override other configurations of the ipcache.

Reply:
> this could potentially cause the identity of the local host to be considered as remote-node which could be bad for policy (denying traffic when it should be allowed)

And skipping the host firewall's policy enforcement :scream_cat: